### PR TITLE
convert modules to use testclusters

### PR DESCRIPTION
**Pull request by alpar-t**
With this change, testclusters gains support for modules and modules use it now.

We also auto configure dependent modules on testclsuters now, no need to make that explicit.

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]
@atorok forgive all my changes but when I ran into `PluginPropertiesTask` during my review I simply had to act 😄 

There was a ton of odd coupling between `PluginBuildPlugin` and that task due to it "owning" the `PluginPropertiesExtension`. This made a bunch of logic in the plugin odd, hard to follow, and not very type safe. I've ditched the task entirely and replaced it with a plain `Copy` task.

I pushed a number of other trivial changes too. Please let me know if you have any questions on any of my changes.

**Comment by mark-vieira** [comment]
I just now realize some of this was addressed in https://github.com/elastic/elasticsearch/pull/39605.

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by atorok** [comment]
thanks @mark-vieira the changes LGTM, and fully agree this is an area that really needed to be cleaned up. We do have others too, and I wanted to keep the PR focused and smaller to make the review easier.  In this case it was easier as I only had to look at your changes :)
